src/gpio_monitor.cpp

```diff
@@ -23,7 +27,37 @@ void GpioEventHandler::setEventHandlerForGpioPresence(
 void GpioMonitor::initHandlerForGpio(
     const std::shared_ptr<boost::asio::io_context>& i_ioContext)
 {
-    // ToDo Add implementation.
-    (void)i_ioContext;
+    if (!m_sysCfgJsonObj.contains("frus"))
+    {
+        logging::logMessage("Frus section is missing in config JSON");
+        return;
+    }
+
+    try
+    {
+        for (const auto& l_Fru : m_sysCfgJsonObj["frus"].items())
+        {
+            const auto l_fruPath = l_Fru.key();
+
+            if (jsonUtility::isPollingRequired(m_sysCfgJsonObj, l_fruPath))
+            {
+                types::GpioPollingParameters l_pollingParameters =
+                    jsonUtility::getGpioPollingParameters(m_sysCfgJsonObj,
+                                                          l_fruPath);
+
+                std::shared_ptr<GpioEventHandler> l_gpioEventHandlerObj =
```

Comment:
The current list of functions looks ok but IMO it would be better if we can abstract maximum json driven functionality to json utiliity?

I would suggest have a base function in json utility, something like 
getFrusParamThatNeedsPolling() 
"Name just a suggestion, please see if you can come up with a better one"

This class should call that function from json utility, and that function should do all the internal processing like calling "getGpioPollingParameters" or "isPollingRequired" and return a list of "GpioPollingParameters" back to this class which can be used to instantiate event handler.

Please let me know your thoughts on this.